pkg/util/timeutil/timeutil.go

```diff
@@ -267,13 +268,19 @@ func FormatDurationStringDaysToHours(param string) (string, error) {
 // RoundToStartOfWeek creates a new time.Time for the preceding Monday 00:00 UTC
 func RoundToStartOfWeek(t time.Time) time.Time {
 	date := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, time.UTC)
+	if date.Weekday() == time.Sunday {
+		return date
+	}
 	daysFromMonday := int(date.Weekday())
 	return date.Add(-1 * time.Duration(daysFromMonday) * Day)
 }
 
 // RoundToStartOfFollowingWeek creates a new time.Time for the following Monday 00:00 UTC
```

Comment:
should we update these comments/var names? Weekday() has the following underlying implementation: 
```
const (
	Sunday Weekday = iota
	Monday
	Tuesday
	Wednesday
	Thursday
	Friday
	Saturday
)
```
... I think this is returning sunday dates, not mondays?

Reply:
Another good catch! Bad naming on my part, though I believe the functionality is correct (will validate + add tests).